test/e2e/main_test.go

```diff
@@ -0,0 +1,407 @@
+//go:build integ
+// +build integ
+
+package kmesh
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path"
+	"path/filepath"
+	"runtime"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/config/schema/gvk"
+	istioKube "istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/ambient"
+	kubecluster "istio.io/istio/pkg/test/framework/components/cluster/kube"
+	"istio.io/istio/pkg/test/framework/components/crd"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/common/ports"
+	"istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/echo/match"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/helm"
+	testKube "istio.io/istio/pkg/test/kube"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/tests/integration/security/util/cert"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gateway "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+var (
+	i istio.Instance
+
+	// KmeshSrc is the location of Kmesh source.
+	KmeshSrc = getDefaultKmeshSrc()
+
+	apps = &EchoDeployments{}
+)
+
+type EchoDeployments struct {
+	// Namespace echo apps will be deployed
+	Namespace namespace.Instance
+
+	// AllWaypoint is a waypoint for all types
+	AllWaypoint echo.Instances
+	// WorkloadAddressedWaypoint is a workload only waypoint
+	WorkloadAddressedWaypoint echo.Instances
+	// ServiceAddressedWaypoint is a service only waypoint
+	ServiceAddressedWaypoint echo.Instances
```

Comment:
I am not sure this is suitable for kmesh, at least the names and comments are confusing

Reply:
removed, we don't need these fields now.